accepted/2020/better-obsoletion/obsoletions-in-net5.md

```diff
@@ -0,0 +1,234 @@
+# Obsoletions in .NET 5
+
+**PM** [Immo Landwerth](https://github.com/terrajobst)
+
+For .NET 5, [we're making](better-obsoletion.md) it more viable to obsolete APIs.
+In this document, we're describing which APIs we intend to obsolete in .NET 5 and
+which diagnostic IDs we will use for those.
+
+With `dotnet/runtime` PR [#33248](https://github.com/dotnet/runtime/pull/33248),
+the `ObsoleteAttribute` gained new `DiagnosticId` and `UrlFormat` properties. With
+these properties, obsoletions can be grouped together when multiple APIs are
+marked as obsolete for the same reason. This capability is reflected in the list
+of obsoletions below. All of these obsoletions will use the `UrlFormat` of
+`https://aka.ms/dotnet-warnings/{0}`.
+
+1. The Constrained Execution Region (CER) feature is no longer implemented
+    * DiagnosticId: `SYSLIB0004`
+    * APIs:
+        * `System.Runtime.ConstrainedExecution.PrePrepareMethodAttribute`
+        * `System.Runtime.ConstrainedExecution.ReliabilityContractAttribute`
+        * `System.Runtime.ConstrainedExecution.Cer`
+        * `System.Runtime.ConstrainedExecution.Consistency`
+        * `System.Runtime.CompilerServices.RuntimeHelpers.ExecuteCodeWithGuaranteedCleanup`
+        * `System.Runtime.CompilerServices.RuntimeHelpers.PrepareConstrainedRegions`
+        * `System.Runtime.CompilerServices.RuntimeHelpers.PrepareConstrainedRegionsNoOP`
+        * `System.Runtime.CompilerServices.RuntimeHelpers.PrepareContractedDelegate`
+        * `System.Runtime.CompilerServices.RuntimeHelpers.ProbeForSufficientStack`
+1. The Global Assembly Cache is no longer implemented
+    * DiagnosticId: `SYSLIB0005`
+    * APIs:
+        * `System.Reflection.Assembly.GlobalAssemblyCache`
+1. `Thread.Abort` is no longer supported and throws PlatformNotSupportedException
+    * DiagnosticId: `SYSLIB0006`
+    * APIs:
+        * `System.Threading.Thread.Abort()`
+        * `System.Threading.Thread.Abort(Object)`
+1. The default implementations of these cryptography algorithms are no longer supported and throw PlatformNotSupportedException
+    * DiagnosticId: `SYSLIB0007`
+    * APIs:
+        * `System.Security.Cryptography.SymmetricAlgorithm.Create()`
+        * `System.Security.Cryptography.AssymetricAlgorithm.Create()`
+        * `System.Security.Cryptography.HMAC.Create()`
+        * `System.Security.Cryptography.KeyedHashAlgorithm.Create()`
```

Comment:
@bartonjs `HashAlgorithm.Create()` also throws (and there's a unit test including it in the same list as these others). Should we obsolete it as well?

Reply:
> HashAlgorithm.Create() also throws (and there's a unit test including it in the same list as these others). Should we obsolete it as well?

Yep.